ci/prow/boskos/create_projects.sh

```diff
@@ -16,18 +16,25 @@
 
 set -e
 
-readonly FIRST=${1:?"First argument is the first number of the new project(s)."}
-readonly NUMBER=${2:?"Second argument is the number of new projects."}
-readonly BILLING_ACCOUNT=${3:?"Third argument must be the billing account."}
-readonly OUTPUT_FILE=${4:?"Fourth argument should be a file name all project names will be appended to in a resources.yaml format."}
+readonly NUMBER=${1:?"First argument is the number of new projects to create."}
+readonly BILLING_ACCOUNT=${2:?"Second argument must be the billing account."}
 
+readonly RESOURCE_FILE="resources.yaml"
+
+start=$(grep -n "knative-boskos-" ${RESOURCE_FILE} | wc -l)
+((start++))
 for (( i=0; i<${NUMBER}; i++ )); do
-  PROJECT="knative-boskos-$(( i + ${FIRST} ))"
+  PROJECT="knative-boskos-$(( ${start} + i ))"
   # This Folder ID is google.com/google-default
   # If this needs to be changed for any reason, GCP project settings must be updated.
   # Details are available in Google's internal issue 137963841.
   gcloud projects create ${PROJECT} --folder=396521612403
   gcloud beta billing projects link ${PROJECT} --billing-account=${BILLING_ACCOUNT}
+
+  # Set permissions for this project
   "$(dirname $0)/set_permissions.sh" ${PROJECT}
-  echo "  - ${PROJECT}" >> ${OUTPUT_FILE}
+
+  last_project_line=$(grep -n "knative-boskos-" ${RESOURCE_FILE} | tail -n 1 | cut -d: -f1)
+  ((last_project_line++))
```

Comment:
This is dangerous, as a change to the order of the lines in the yaml file will break this calculation. Instead, grep the last entry and use sed's `/a` to add after the match.

Reply:
Ditto